### ISSUE TITLE
All printers show up as "not available right now"

### ISSUE THREAD

**Issue by speachy** [BenoitDuffez/AndroidCupsPrint#146]
I am seeing this with v1.5.0 via Play Store, and also a copy I compiled myself.  Running on an Android 8.0 device.

First, I have 20 printers configured on the server, but only 18 show up in the discovery list.  But that's not the reason for this ticket.   I am able to print from other (Linux) clients fine via both ipv4 and ipv6, so the IPP discovery/printing path on the server is fundamentally functional.

The problem I'm seeing is that no matter which printer I try to select, I get this error popping up:

"Printer not responding.  Please check it is powered on and available at http://..../printers/printername"

And on the main screen it states "This printer isn't available right now."

Looking at the CUPS logs it appears as if specific printer URIs are never being queried; after the "CUPS-Get-Default" query the client closes the connection.

**Comment by diggy128** [comment]
Same thing here too.
It's a shame you've put so much time and can't get the app to print.
Any change to triage this? It a killer app to have for those on the road.

**Comment by BenoitDuffez** [comment]
Yeah I'm sorry I just don't have time to troubleshoot this anymore. Also the app seems to be fine to most so my time spent is not lost ;)

**Comment by diggy128** [comment]
I'm glad it works for most.
BTW if you find time someday, I'd be willing to donate some money to help fix this.

**Comment by tenzap** [comment]
Hello, I faced the same problem.

To fix it, this is what I did:
In the CUPS administration panel @ http://127.0.0.1:631/admin I checked this option:
`Allow remote administration`

The printer was then available through this app (message that printer is not available is gone) , but they were also detected/available through the built-in printer service

Edit: Actually it is sufficient to configure CUPS as described in the section [Sharing via Internet Printing Protocol](https://wiki.archlinux.org/index.php/CUPS/Printer_sharing#Sharing_via_Internet_Printing_Protocol)

My phone uses Android 10

**Comment by speachy** [comment]
FWIW I have remote admin turned on and with the latest version out of the Play store, I'm still unable to discover printers most of the time, much less successfully print anything (Android 9 client)

I'm able to print over the network from multiple clients just fine; it's just Android that's pathological.  Even my parner's iPhone works fine with this CUPS server, and airprint is ... special most of the time.

**Comment by tenzap** [comment]
Well, [that](https://android.stackexchange.com/a/230116) worked for me with just the android stuff. I'm sorry it doesn't for you.
